homeassistant/components/zone/__init__.py

```diff
@@ -57,10 +57,12 @@
             zone.entity_id = async_generate_entity_id(
                 ENTITY_ID_FORMAT, entry[CONF_NAME], None, hass)
             hass.async_add_job(zone.async_update_ha_state())
-            hass.data[DOMAIN][name] = zone
+            hass.data[DOMAIN][slugify(name)] = zone
 
     if HOME_ZONE not in hass.data[DOMAIN] and HOME_ZONE not in zone_entries:
         name = hass.config.location_name
+        if name is None:
```

Comment:
It's possible apply a proper default to `location_name`?

Reply:
That is hass own parameter. This is how the old logic worked. I just missed this use case